Description: Fix CVE-2025-67221 sporadic crash on serializing object close
 The orjson.dumps function would crash sporadically when serializing
 deeply nested JSON documents. The issue was that end_object() used
 debug_assert_has_capacity!() which only checks buffer capacity in debug
 builds but does nothing in release builds. This patch changes it to
 reserve_minimum!() which actually reserves buffer space, preventing
 potential buffer overruns.
Origin: upstream, https://github.com/ijl/orjson/commit/62bb185b70785ded49c79c26f8c9781f1e6fe370
Author: ijl
Last-Update: 2026-02-04

--- a/src/serialize/writer/formatter.rs
+++ b/src/serialize/writer/formatter.rs
@@ -244,7 +244,7 @@ pub trait Formatter {
     where
         W: ?Sized + io::Write + WriteExt,
     {
-        debug_assert_has_capacity!(writer);
+        reserve_minimum!(writer);
         unsafe {
             writer.write_reserved_punctuation(b'}').unwrap();
         }
